core/src/main/java/hudson/model/Descriptor.java

```diff
@@ -984,7 +984,14 @@ private URL getStaticHelpUrl(Klass<?> c, String suffix) {
     Map<Descriptor<T>,T> toMap(Iterable<T> describables) {
         Map<Descriptor<T>,T> m = new LinkedHashMap<Descriptor<T>,T>();
         for (T d : describables) {
-            m.put(d.getDescriptor(),d);
+            Descriptor<T> descriptor;
+            try {
+                descriptor = d.getDescriptor();
+            } catch (Throwable x) {
```

Comment:
Suppressing all throwables is generally a bad practice, but we do not have much choice since some Jenkins extension types implement Throwable directly

Reply:
Yes, unfortunately the throw of `AssertionError` is baked into `getDescriptorOrDie`, yet we have no way of knowing if a particular `getDescriptor` implementation is going to call that or not.